File: javascript/1544.Make-The-String-Great.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function (s) {
  const stack = [];

  for (let i = 0; i < s.length; i++) {
    if (
      stack.length &&
      s[i] !== stack[stack.length - 1] &&
      s[i].toLowerCase() === stack[stack.length - 1].toLowerCase()
    ) {
      stack.pop();
    } else {
      stack.push(s[i]);
    }
  }

  return stack.join("");
};

File: python/1544.Make-The-String-Great.py
class Solution:
    def makeGood(self, s: str) -> str:
        stk = []

        for i in range(len(s)):
            if len(stk) and s[i] != stk[-1] and s[i].lower() == stk[-1].lower():
                stk.pop()
            else:
                stk.append(s[i])

        return "".join(stk)